### PR TITLE
Add cardholder name

### PR DESCRIPTION
**Pull request by mgroger2**

### CODE REVIEW

File: CardIOCardholderNameTextFieldDelegate.m [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#import "CardIOCardholderNameTextFieldDelegate.h"
+
+@implementation CardIOCardholderNameTextFieldDelegate
+
+-(id) init {
+  if ((self = [super init])) {
+    // Globalization: alphanumeric, space, hyphen are all definitely okay;
+    // there's no compelling reason for us to get fussy here.
+    self.numbersOnly = NO;
+    self.maxLength = 175;  // PayPal REST APIs accept max of 175 chars for cardholder name
```
Comment (martinrybak) [inline CardIOCardholderNameTextFieldDelegate.m:15]: Should this be defined as a constant?

File: CardIOCardholderNameTextFieldDelegate.m [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+//  CardIOCardholderNameTextFieldDelegate.m
+//  See the file "LICENSE.md" for the full license governing this code.
+//
+
+#import "CardIOCardholderNameTextFieldDelegate.h"
+
+@implementation CardIOCardholderNameTextFieldDelegate
+
+-(id) init {
+  if ((self = [super init])) {
```
Comment (martinrybak) [inline CardIOCardholderNameTextFieldDelegate.m:11]: The extra set of parentheses is not strictly required.

File: CardIOCardholderNameTextFieldDelegate.m [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    // Globalization: alphanumeric, space, hyphen are all definitely okay;
+    // there's no compelling reason for us to get fussy here.
+    self.numbersOnly = NO;
+    self.maxLength = 175;  // PayPal REST APIs accept max of 175 chars for cardholder name
+  }
+  return self;
+}
+
++(BOOL)isValidCardholderName:(NSString*)cardholderName {
+  return [cardholderName length] > 0;
```
Comment (martinrybak) [inline CardIOCardholderNameTextFieldDelegate.m:21]: Where does it check for max length?

File: Classes/CardIODataEntryViewController.m [hunk 4]
```diff
@@ -929,12 +968,29 @@ - (void)postalCodeDidChange:(id)sender {
   [self validate];
 }
 
+- (void)cardholderNameDidChange:(id)sender {
+  self.cardInfo.cardholderName = self.cardholderNameTextField.text;
+
+  if([CardIOCardholderNameTextFieldDelegate isValidCardholderName:self.cardInfo.cardholderName]) {
+    [self advanceToNextEmptyFieldFrom:self.cardholderNameTextField];
+    self.cardholderNameTextField.textColor = [CardIOTableViewCell defaultDetailTextLabelColorForCellStyle:[CardIOTableViewCell defaultCellStyle]];
+  } else if(self.cardholderNameTextField.text.length >= 175) {
```
Comment (martinrybak) [inline Classes/CardIODataEntryViewController.m:977]: This 175 was also defined in `CardIOCardholderNameTextFieldDelegate.m`. Is there a way to reuse?

File: Classes/CardIODataEntryViewController.m [hunk 5]
```diff
@@ -415,6 +445,12 @@ - (void)viewDidAppear:(BOOL)animated {
                                                  name:UITextFieldTextDidChangeNotification
                                                object:self.postalCodeTextField];
   }
+  if(self.collectCardholderName) {
```
Comment (martinrybak) [inline Classes/CardIODataEntryViewController.m:448]: `CardIOCardholderNameTextFieldDelegate` is used in conjunction with `UITextFieldTextDidChangeNotification`?

File: CardIOCardholderNameTextFieldDelegate.m (hunk 2 above)
Comment (mgroger2) [inline CardIOCardholderNameTextFieldDelegate.m:11]: That's their style on all of the other textFieldDelegate files.

File: CardIOCardholderNameTextFieldDelegate.m (hunk 1 above)
Comment (mgroger2) [inline CardIOCardholderNameTextFieldDelegate.m:15]: I agree that it should be a constant, but, again, this is strictly following their style (of not using constants). I think I'll do what Dan did, and after this branch gets merged, make another PR for Card.io that cleans up their code and implements constants.

File: Classes/CardIODataEntryViewController.m (hunk 4 above)
Comment (mgroger2) [inline Classes/CardIODataEntryViewController.m:977]: They define the max length for Expiry as 7 in both of the same places. I figured following their style was better than being potentially confusing with more organized code.

**Pull request merged**